### PR TITLE
[13.0][MIG] im_livechat

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Migration of `im_livechat`.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/im_livechat/migrations/13.0.1.0/openupgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+im_livechat  / im_livechat.channel      / rating_ids (one2many)         : NEW relation: rating.rating
+im_livechat  / mail.channel             / country_id (many2one)         : NEW relation: res.country
+im_livechat  / res.users                / livechat_username (char)      : NEW
+# NOTHING TO DO (tentative)
+
+im_livechat  / mail.channel             / livechat_operator_id (many2one): NEW relation: res.partner
+# DONE: pre-migration: filled with the user_ids of im_livechat.channel
+
+---XML records in module 'im_livechat'---
+NEW im_livechat.channel: im_livechat.im_livechat_channel_data
```
Comment (kos94ok-3D) [inline addons/im_livechat/migrations/13.0.1.0/openupgrade_analysis_work.txt:18]: renamed from `website_livechat.im_livechat_channel_data_website`
https://github.com/odoo/odoo/commit/318893d6ed941f81bbbea230a087a3574cc97dcb

**Review by kos94ok-3D — CHANGES_REQUESTED** [review]

**Review by kos94ok-3D — APPROVED** [review]

File: addons/im_livechat/migrations/13.0.1.0/pre-migration.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 ForgeFlow <http://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+_field_renames = [
+    ('im_livechat.channel', 'im_livechat_channel',
```
Comment (pedrobaeza) [inline addons/im_livechat/migrations/13.0.1.0/pre-migration.py:7]: You need to do more than renaming the field, as this is an attachment field, so there will be attachments with the incorrect `res_field`, so you will lose all the existing ones.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: addons/im_livechat/migrations/13.0.1.0/pre-migration.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply addons/im_livechat/migrations/13.0.1.0/pre-migration.py:7]: but I think that is done in the method below :S

Comment (pedrobaeza) [reply addons/im_livechat/migrations/13.0.1.0/pre-migration.py:7]: You are right, sorry. I didn't remember we included such condition.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**